src/streaming/controllers/BufferController.js

```diff
@@ -131,16 +129,15 @@ function BufferController(config) {
         eventBus.on(Events.PLAYBACK_RATE_CHANGED, onPlaybackRateChanged, this);
         eventBus.on(Events.PLAYBACK_SEEKING, onPlaybackSeeking, this);
         eventBus.on(Events.WALLCLOCK_TIME_UPDATED, onWallclockTimeUpdated, this);
-        eventBus.on(Events.CURRENT_TRACK_CHANGED, onCurrentTrackChanged, this);
+        eventBus.on(Events.CURRENT_TRACK_CHANGED, onCurrentTrackChanged, this, 0);
```

Comment:
If we do something like states of priority CONST  the question is, is that scalable?  What happens if there are five preset states of priority but then the use of an event type grows to 6 listeners and you need all six in a certain order.  I opted for a pure int based system which is limitless so I want to see how we can make this more clear as I do agree this needs some help.   However in most other event systems I have used it is the reverse,  Higher number holds more priority.